CHANGELOG.md

```diff
@@ -1,3 +1,76 @@
+# 4.0 (alpha-1)
+
+If you're new to the Swift API and migrating from RAC 2, start with the
+[3.0 changes](#30). This section only covers the differences when between 3.0
+and 4.0.
+
+ReactiveCocoa 4.0 targets Swift 2 and the current focus is on leveraging the
+improvements from Swift 1.2 to provide a simpler API.
+
+#### Signal operators are protocol extensions
+
+The biggest change from RAC 3 to RAC 4 is that signal and producer operators
+are implemented as protocol extensions instead of global functions. This is
+similar to many of the collection protocol changes in the Swift 2 standard
+library.
+
+This enables chaining signal operators with normal dot-method calling syntax.
+Previously the custom `|>` was required to enable chaining global functions
+without a mess of nested calls and parenthesis.
+
+```swift
+/// RAC 3
+signal |> filter { $0 % 2 == 0 } |> map { $0 * $0 } |> observe { print($0) }
+
+/// RAC 4
+signal.filter { $0 % 2 == 0 } .map { $0 * $0 } .observe { print($0) }
+```
+
+Additionally, this means that `SignalProducer` operators are less "magic". In
+RAC 3 the `Signal` operators were implicitly lifted to work on `SignalProducer`
+via `|>`. This was a point of confusion for some, especially when browsing the
+source looking for these operators. Now as protocol extensions, the
+`SignalProducer` operators are explicitly implementated in terms of their
+`Signal` counterpart when available.
+
+#### Removal of |> custom operator
+
+As already alluded to above, the custom `|>` operator for chaining signals has
+been removed. Instead standard method calling syntax is used for chaining
+operators.
+
+#### Event.Sink is now a function
+
+With the removal of `SinkType` in Swift 2, the `Event.Sink` type is now just a
+function `Event -> ()`.
+
+#### Event cases are no longer boxed
+
+The improvements to associated enum values in Swift 2 mean that `Event` cases
+no longer need to be `Box`ed. In fact, the `Box` dependency has been removed
+completed from RAC 4.
+
+#### Replacements for the start and observer overloads
+
+The `observe` and `start` overloads taking `next`, `error`, etc. optional
+function parameters have been removed. This was necessitated by the change to
+`Event.Sink` becoming a function type which introduced an unresolvable
+ambiguity. They've been replaced with methods taking a single function with
+the target `Event` case -- `observeNext`, `startWithNext`, and the same for
+error and completed. See #2311 and #2318 for more details.
```

Comment:
After attempting to use this new style in practice (upgrading [neilpa/Rex](https://github.com/neilpa/rex)), I'm not a fan. I think we need another pass at this change, potentially bringing back the `start` overload but with `interrupted` as the last param instead of `next`. This is one place where trailing closure syntax is getting in the way.